src/lib/openapi/spec/environment-project-schema.ts

```diff
@@ -5,29 +5,51 @@ export const environmentProjectSchema = {
     type: 'object',
     additionalProperties: false,
     required: ['name', 'type', 'enabled'],
```

Comment:
So here's a thing to discuss. The way OpenAPI works, **all** properties that aren't specified in the `required` list are considered optional. From [the OpenAPI docs](https://swagger.io/docs/specification/data-models/data-types/#required):

>  By default, all object properties are optional. You can specify the required properties in the required list: 

That means, they can be undefined (as in: not included in the object). My guess is that most of the other fields on this schema aren't optional: they're always there. Can we specify all the properties that aren't optional in this list. And if a property really is optional, then its description should explicitly state when it will be missing. What do you think?

Reply:
I think you're right, since this is a response schema, not a request schema, it should be safe to mark them all as required. I'll update the PR

Reply:
Hmm, interesting discussion. The types we're returning here has these fields set as optional, so should we guarantee that they are here using required and change the Typescript type as well?

Reply:
I see we have a call for Environments which does not return counts.

Reply:
I'm guessing this is an instance of reusing types multiple places, where what we'd like is at least two different types, one with counts and one without, instead of trying to reuse the same type by making the fields optional